subsys/logging/backends/log_backend_adsp_mtrace.c

```diff
@@ -221,7 +221,7 @@ void adsp_mtrace_log_init(adsp_mtrace_log_hook_t hook)
 	mtrace_init();
 
 	mtrace_hook = hook;
-	mtrace_active = true;
+	mtrace_active = !!hook;
```

Comment:
Please make it explicit.

```suggestion
	mtrace_active = hook != NULL;
```

Reply:
@pdgendt is this a Zephyr style requirement? Could you point me to the relevant text?

Reply:
Yes MISRA-C 2012 rule 10.4 if the [coding guidelines](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html). I would also argue readability.

Reply:
@pdgendt the one in line 62 in the table? I'm not sure it's the correct one here - it's talking about "both operands" i.e. it's about binary operators, while here only unary negation is used. Readability is subjective IMHO.

Reply:
You're correct, so it will be rule 10.5 which is "only advisory", as can be seen from the examples:
```c
   bool_t bna = ( bool_t ) false;  /* Compliant - C99 'false' is essentially Boolean */
   use_bool ( bna );
   bna = ( bool_t ) 0;        /* Compliant  - by exception */
   use_bool( bna );
   bna = ( bool_t ) 3U;       /* Non-compliant  */
   use_bool( bna );
```

But I would argue that this is similar to rule 14.4, which IS required and where `if (hook)` wouldn't be allowed, so
`if (!!hook)` shouldn't be either.

Reply:
In any case, I'm not blocking this PR.